src/game-engine/GameEngine.ts

```diff
@@ -154,36 +147,34 @@ export default class GameEngine {
         opponentMessage: message,
       });
     } else if (stateType === ApplicationStatesA.types.WaitForAccept) {
-      const result = RpsGame.result(oldState.aPlay.key, opponentState.bPlay.key);
+      const { aPlay, salt, stake} = oldState;
+      const bPlay = opponentState.bPlay;
+      const turnNum = opponentState.turnNum;
+      const result = calculateResult(aPlay, bPlay);
 
       // The opponent's state assumes that B won
-      const { resolution } = opponentState;
-      if (result === RpsGame.Results.TIE) {
-        resolution[0] += opponentState.stake;
-        resolution[1] -= opponentState.stake;
-      } else if (result === RpsGame.Results.A) {
-        resolution[0] += 2 * opponentState.stake;
-        resolution[1] -= 2 * opponentState.stake;
+      let newBalances = balances;
```

Comment:
I think this will only create a shallow copy - probably want to use `[...balances]` for safety.